src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTaggerProvider.cs

```diff
@@ -16,25 +18,39 @@ namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
     /// The provider that is used as a middleman to create the tagger so that the data tag
     /// can be used to create the UI tag
     /// </summary>
-    [Export(typeof(ITaggerProvider))]
+    [Export(typeof(IViewTaggerProvider))]
     [ContentType(ContentTypeNames.RoslynContentType)]
     [TagType(typeof(IntraTextAdornmentTag))]
     [Name(nameof(InlineParameterNameHintsTaggerProvider))]
-    internal class InlineParameterNameHintsTaggerProvider : ITaggerProvider
+    internal class InlineParameterNameHintsTaggerProvider : IViewTaggerProvider
     {
-        private readonly IBufferTagAggregatorFactoryService _bufferTagAggregatorFactoryService;
+        private readonly IViewTagAggregatorFactoryService _viewTagAggregatorFactoryService;
+        public readonly IClassificationFormatMapService ClassificationFormatMapService;
+        public readonly IClassificationTypeRegistryService ClassificationTypeRegistryService;
+        public readonly IThreadingContext ThreadingContext;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public InlineParameterNameHintsTaggerProvider(IBufferTagAggregatorFactoryService bufferTagAggregatorFactoryService)
+        public InlineParameterNameHintsTaggerProvider(IViewTagAggregatorFactoryService viewTagAggregatorFactoryService,
+                                                       IClassificationFormatMapService classificationFormatMapService,
+                                                       IClassificationTypeRegistryService classificationTypeRegistryService,
+                                                       IThreadingContext threadingContext)
         {
-            _bufferTagAggregatorFactoryService = bufferTagAggregatorFactoryService;
+            _viewTagAggregatorFactoryService = viewTagAggregatorFactoryService;
+            this.ClassificationFormatMapService = classificationFormatMapService;
+            this.ClassificationTypeRegistryService = classificationTypeRegistryService;
+            this.ThreadingContext = threadingContext;
         }
 
-        public ITagger<T> CreateTagger<T>(ITextBuffer buffer) where T : ITag
+        public ITagger<T> CreateTagger<T>(ITextView textView, ITextBuffer buffer) where T : ITag
         {
-            var tagAggregator = _bufferTagAggregatorFactoryService.CreateTagAggregator<InlineParameterNameHintDataTag>(buffer);
-            return new InlineParameterNameHintsTagger(buffer, tagAggregator) as ITagger<T>;
+            if (buffer != textView.TextBuffer)
```

Comment:
Please add a comment about what this check is trying to accomplish and why check works.

Reply:
Done